sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -92,6 +92,9 @@ case class AdaptiveSparkPlanExec(
   // optimizations should be stage-independent.
   @transient private val queryStageOptimizerRules: Seq[Rule[SparkPlan]] = Seq(
     ReuseAdaptiveSubquery(conf, subqueryCache),
+    // Here we need put the OptimizeLocalShuffleReader rule before
+    // ReduceNumShufflePartitions rule to avoid the further optimizaiton.
```

Comment:
I think the comment needs to explain 2 things:
1. why execute this rule twice
2. why it must be run before `OptimizeLocalShuffleReader`

Reply:
Updated.